### PR TITLE
Proposed edits to Lesson 2

### PR DESCRIPTION
**Pull request by canyon289**
Hey Tomas. This script is awesome. Theres a lot of pieces between the code, story, pictures, narrative etc so just know if this feels like a lot of comments from me its in the nominal amount.\ There's three major changes I made here that I'll list

Most everything I say below is a proposal and doesn't have to be implemented exactly, or at all

## Added jupytext
This is a technical note. Jupytext makes it easy to diff the notebooks. We dont need to use jupytext at all, I just did it at least temporarily for this PR to make some comments simpler

```pip install jupytext```

## Revised the introduction
The intent and tone of the introduction section were spot on. You start with motivating the students with a story, then explain what were going to do. What I changed primarily is
* Condensing the story into three slides
* Adding an outline of what is to come in the lesson

The reason I did this is in the intro course when we had overly long intros people tended to leave feedback about it. The second part was adding in an outline of what were going to do specifically. This gives students a mental map of what theyll learn in the lesson. Now I know this is bit redundant with then us making a plan in EDA section, but I actually think it might be ok because in the EDA Section we show the how we got to the plan, not just that we as the instructors "knew" the plan. 

## Consolidated note slides
In a couple areas you had this. 
```
* Slide
* Notes 1
* Content
* Notes 2
```
The issue is in presenter view only the first notes get shown. I changed it those blocks to be like this so everything works

```
* Slide
* Content
*  Notes 1 and Notes 2
```

## Changed everything to h1 and h2 and added titles
Added titles to add slides. The reason for this is consistency and so the headers are large enough to read

## Reduced onscreen text
There were lots of comments in introductory course about text heavy slides. Theyre challenging because its hard to read and listen at the same time. The other thing to note is we have captions, so when we say the thing its both more fun to hear, people can read it anyway, **and its searchable later**. So if people want to see where we talk about correlation for instance, they can search the captions to find the exact second we do which is cool

## Things remaining
* Some pictures can be resized to fit the screen better. This is easy
* Maybe some narrative smoothing. Not much
* **I didnt edit anything in worlds simplest linear model section**

### CODE REVIEW

**Comment by tomicapretto** [comment]
@canyon289 thanks a lot for the prompt feedback! I haven't reviewed everything yet, but I'll post comments as I review.

image
I wasn't aware of this, but now that you raise it, it makes sense to have non-long intros. I think "making a plan" after EDA is the right thing to do, so it doesn't feel it's magical. I also think there could be an initial more general description of the lesson in the very beginning (problem introduction, EDA, modeling, conclusions) and a more specific description for the modeling, which is what we already have in the last part of the EDA section.

image

I was aware of this when doing it. I was thinking to have another screen with the notes when recording. This is because in many places it felt weird to have all the notes below the content. In many cases, the notes were an introduction to the content that followed. Haven't seen the changes yet. But if the changes accommodate this in a way everything fits, I'm happy to change that.

image
Thanks for it!

image
This is a good point. It was an open question for me until now. So, the goal would be to have less text on the slides, but that doesn't necessarily mean less text on the notebooks, right?

File: lessons/lesson_02.ipynb [hunk 1]
```diff
@@ -22,7 +22,7 @@
    },
```
Comment (tomicapretto) [inline lessons/lesson_02.ipynb:22]: <pre class="ql-syntax" spellcheck="false">Line #1.    import arviz as azt
</pre><p>I guess this is a typo</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tomicapretto/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (tomicapretto) [inline lessons/lesson_02.ipynb:22]: <p>I think these are really good suggestions. If I understand correctly, you're not saying that we don't have to introduce the problem saying we are data scientists at a company that sell fish. You're just not making that a priority in the slides, right?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tomicapretto/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (tomicapretto) [inline lessons/lesson_02.ipynb:22]: <p>This way of writing makes a huge difference. Thanks!!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tomicapretto/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (canyon289) [reply lessons/lesson_02.ipynb:22]: Id think of it more as a rebalancing the background story versus the lesson syllabus. Its one of the trickiest things to get right and even now Im not sure if its the right balance. Certainly will take a little bit more discussion and a couple more iterations

**Comment by canyon289** [comment]
> I was aware of this when doing it. I was thinking to have another screen with the notes when recording. This is because in many places it felt weird to have all the notes below the content. In many cases, the notes were an introduction to the content that followed. Haven't seen the changes yet. But if the changes accommodate this in a way everything fits, I'm happy to change that.

The goal I'd think about is "How can we get the best possible video?" Remember students will be seeing this content through video primarily, and notebook secondarily (if at all). If the current note structure allow you to do that then I have no issues. Once you start recording lessons you'll know whatll get you the best end product. If the current structure is what does it then lets keep it!

>  This is a good point. It was an open question for me until now. So, the goal would be to have less text on the slides, but that doesn't necessarily mean less text on the notebooks, right?

Correct, this doesnt necessarily mean less text in the notebook, though know we also dont want overly long scripts, where we stay on a single slide and talk for over a minute. When you think about text think of it in these three categories

1. Text thats visible the slide and therefore the video
2. Text thats a  script for us, which is part of the performance
3. Extra reference text in the notebooks thats not read on the slides but meant for students to read as reference if they want

File: lessons/lesson_02.ipynb (hunk 1 above)
Comment (canyon289) [reply lessons/lesson_02.ipynb:22]: Ah yes my apologies ill fix it

**Comment by canyon289** [comment]
Merging this one as weve made so many changes, and with another PR coming for more itd get confusing. If this was preliminary we can always roll back

**Pull request merged**